packages/core/src/stdlib/typeHelpers.ts

```diff
@@ -18,8 +18,11 @@ import { NullObject } from "../runtime/objects/nullObject.js";
  * @param description the description of the value, part of the error message
  * @returns the value of the StringObject
  */
-export function assertString(value: BaseObject, description = ""): string {
+export function assertString(value: BaseObject, description: string | (() => string) = ""): string {
```

Comment:
maybe
- update the param description
- extract a type for this
- update the other assert functions as well